.github/workflows/coverity_scan.yml

```diff
@@ -0,0 +1,46 @@
+name: coverity-scan
+on:
+  workflow_dispatch: # because sometimes you just want to force a branch to have tests run
+  push:
+    branches:
+      - "**"
+    paths:
+      - "backend/**"
+      - .github/workflows/coverity_scan.yml
+  schedule:
+    - cron: '0 18 * * *' # Daily at 18:00 UTC
```

Comment:
why is this run on a schedule?

Reply:
A couple of reasons:

1. The analysis takes a while, so it's not something that is particularly useful to run on a per commit basis. Depending on the queue length it can take 20 minutes to an hour or more.
2. We have build limits:

```
The number of weekly builds per project are as follows:

Up to 28 builds per week, with a maximum of 4 builds per day, for projects with fewer than 100K lines of code
Up to 21 builds per week, with a maximum of 3 builds per day, for projects with 100K to 500K lines of code
Up to 14 builds per week, with a maximum of 2 build per day, for projects with 500K to 1 million lines of code
Up to 7 builds per week, with a maximum of 1 build per day, for projects with more than 1 million lines of code
```

The tool itself is designed to be run on a more irregular basis, as such, it tries to prioritize more complex analyses, but at the cost of long scan times.

Reply:
Seems like we should not run on push, then? Or perhaps on push but only for the workflow file?

Reply:
Whoops. Copy/pasta. The goal is to only build when the workflow file changes.